### ISSUE TITLE
test_archive_extract failure on armhf (IndexError: list index out of range)

### ISSUE THREAD

**Issue by sten0** [borgbase/vorta#913]
Hi @m3nu!

Using 0.7.5, informal tests on a mixed armel/armhf machine passed; however, running formal debci on on qemu-emulated pure armhf, I found this:
```
=== Testing with python3.9 ===
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-6.0.2, py-1.10.0, pluggy-0.13.0
PyQt5 5.15.2 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
rootdir: /tmp/autopkgtest-lxc.hadukhki/downtmp/autopkgtest_tmp/vorta
plugins: qt-3.2.2, forked-1.3.0, mock-1.10.4
collected 54 items

tests/test_archives.py .....F............                                [ 33%]
tests/test_borg.py .                                                     [ 35%]
tests/test_lock.py ..                                                    [ 38%]
tests/test_misc.py s                                                     [ 40%]
tests/test_notifications.py .                                            [ 42%]
tests/test_profile.py ..                                                 [ 46%]
tests/test_repo.py ....s.                                                [ 57%]
tests/test_schedule.py .                                                 [ 59%]
tests/test_scheduler.py .                                                [ 61%]
tests/test_source.py .                                                   [ 62%]
tests/test_utils.py .                                                    [ 64%]
tests/network_manager/test_darwin.py ...                                 [ 70%]
tests/network_manager/test_network_manager.py ................           [100%]

=================================== FAILURES ===================================
_____________________________ test_archive_extract _____________________________

qapp = <vorta.application.VortaApp object at 0xb1454ad8>
qtbot = <pytestqt.qtbot.QtBot object at 0xb047add8>
mocker = <pytest_mock.MockFixture object at 0xaf75a760>
borg_json_output = <function borg_json_output.<locals>._read_json at 0xaf76b220>

    def test_archive_extract(qapp, qtbot, mocker, borg_json_output):
        main = qapp.main_window
...back 1 page
=== Testing with python3.9 ===
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-6.0.2, py-1.10.0, pluggy-0.13.0
PyQt5 5.15.2 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
rootdir: /tmp/autopkgtest-lxc.hadukhki/downtmp/autopkgtest_tmp/vorta
plugins: qt-3.2.2, forked-1.3.0, mock-1.10.4
collected 54 items

tests/test_archives.py .....F............                                [ 33%]
tests/test_borg.py .                                                     [ 35%]
tests/test_lock.py ..                                                    [ 38%]
tests/test_misc.py s                                                     [ 40%]
tests/test_notifications.py .                                            [ 42%]
tests/test_profile.py ..                                                 [ 46%]
tests/test_repo.py ....s.                                                [ 57%]
tests/test_schedule.py .                                                 [ 59%]
tests/test_scheduler.py .                                                [ 61%]
tests/test_source.py .                                                   [ 62%]
tests/test_utils.py .                                                    [ 64%]
tests/network_manager/test_darwin.py ...                                 [ 70%]
tests/network_manager/test_network_manager.py ................           [100%]

=================================== FAILURES ===================================
_____________________________ test_archive_extract _____________________________

qapp = <vorta.application.VortaApp object at 0xb1454ad8>
qtbot = <pytestqt.qtbot.QtBot object at 0xb047add8>
mocker = <pytest_mock.MockFixture object at 0xaf75a760>
borg_json_output = <function borg_json_output.<locals>._read_json at 0xaf76b220>

    def test_archive_extract(qapp, qtbot, mocker, borg_json_output):
        main = qapp.main_window
        tab = main.archiveTab
        main.tabWidget.setCurrentIndex(3)
    
        tab.populate_from_profile()
        qtbot.waitUntil(lambda: tab.archiveTable.rowCount() == 2)
    
        tab.archiveTable.selectRow(0)
        stdout, stderr = borg_json_output('list_archive')
        popen_result = mocker.MagicMock(stdout=stdout, stderr=stderr, returncode
=0)
        mocker.patch.object(vorta.borg.borg_thread, 'Popen', return_value=popen_
result)
        tab.list_archive_action()
    
        qtbot.waitUntil(lambda: hasattr(tab, '_window'), **pytest._wait_defaults
)
        # qtbot.waitUntil(lambda: tab._window == qapp.activeWindow(), **pytest._
wait_defaults)
    
>       assert tab._window.treeView.model().rootItem.childItems[0].data(0) == 'U
sers'
E       IndexError: list index out of range

tests/test_archives.py:128: IndexError
----------------------------- Captured Qt messages -----------------------------
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
    QtWarningMsg: QImage::scaleHeight: Image is a null image
=============================== warnings summary ===============================
tests/test_schedule.py: 6 warnings
tests/test_scheduler.py: 3 warnings
tests/test_source.py: 2 warnings
tests/test_utils.py: 1 warning
tests/network_manager/test_darwin.py: 3 warnings
tests/network_manager/test_network_manager.py: 16 warnings
  /usr/lib/python3/dist-packages/apscheduler/schedulers/qt.py:34: DeprecationWar
ning: an integer is required (got type float).  Implicit conversion to integers 
using __int__ is deprecated, and may be removed in a future version of Python.
    self._timer = QTimer.singleShot(wait_time, self._process_jobs)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED tests/test_archives.py::test_archive_extract - IndexError: list index ...
======= 1 failed, 51 passed, 2 skipped, 31 warnings in 595.61s (0:09:55) =======
```

This output was from a formal debci run using the armhf image from https://people.debian.org/~gio/dqib/

I'll contact the ARM-porter team to find out if the mixed armel/armhf success is more significant than the emulated pure armhf failure, and I've reported this early, because time is of the essence right now.

**Comment by m3nu** [comment]
If it's just a slightly slower UI, this may fix it:

https://github.com/borgbase/vorta/pull/914/files

**Comment by m3nu** [comment]
> This output was from a formal debci run using the armhf image from https://people.debian.org/~gio/dqib/

The armhf VM was easy enough to set up. I'm using dependencies from Debian and the `master` branch from Github. Command to start same as in the README: `$ qemu-system-arm -machine virt -cpu cortex-a15 ...`

Tests take a while (macOS host), but couldn't trigger any error after 5 runs

Maybe this test was running an older version? If it's reliable to reproduce, maybe #914 helps?

**Comment by sten0** [comment]
@m3nu, you're right.  I failed to notice that the CI instance was downloading 0.7.4-1 from Debian archives rather than installing the package I passed it as an argument (yes, really what?!).  I'm sorry for the noise, it's wholly my mistake; I panicked thinking "oh no, we can't afford to lose a day", but should have read the test output more carefully.  Proof of success with on virtualised pure armhf attached, so I'm closing the issue as resolved
[vorta_0.7.5-1.buildlog.txt](https://github.com/borgbase/vorta/files/6087878/vorta_0.7.5-1.buildlog.txt)
